apps/sms/views/shared/js/utils.js

```diff
@@ -33,40 +33,93 @@
     },
     getFormattedHour: function ut_getFormattedHour(time) {
       this.date.shared.setTime(+time);
-      return this.date.format.localeFormat(
-        this.date.shared,
-        navigator.mozL10n.get(
-          navigator.mozHour12 ? 'shortTimeFormat12' : 'shortTimeFormat24'
-        )
-      );
+      var formatter = new Intl.DateTimeFormat(navigator.languages, {
+        hour12: navigator.mozHour12,
+        hour: 'numeric',
+        minute: 'numeric',
+      });
+      return formatter.format(this.date.shared);
     },
     getDayDate: function re_getDayDate(time) {
       this.date.shared.setTime(+time);
       this.date.shared.setHours(0, 0, 0, 0);
       return this.date.shared.getTime();
     },
-    getHeaderDate: function ut_giveHeaderDate(time) {
-      var _ = navigator.mozL10n.get;
+    setHeaderDate: function ut_setHeaderDate(time, element, withTime) {
```

Comment:
I don't like booleans parameters; I'd rather use a parameter object here, but we can use ES6 constructs to make it easier to use here:
```
  setHeaderDate({ time, element, withTime }) {
    ...
  }
```